### PR TITLE
MQTT5 Topic Alias

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
Fixes bug discovered which was reason for (https://github.com/netdata/netdata/pull/14145) - fixes are part of last commits so that is clear to reviewers what was changed since (#14067):
- rx topic alias list was not cleared on reconnect upon failure (according to MQTT spec topic aliases are not to be persisted across connection even if same session is continued
- this in combination with us not allowing topic alias reassignment caused agent not being able to reconnect

Reintroduces https://github.com/netdata/netdata/pull/14067

Automated tests are being worked on to prevent such problems in future.

##### Test Plan
This has to be tested extensively as it introduces wide changes to the mqtt_websockets incl. both parsers and message generators.
- test connection is stable
- test connection is eventually established after `ss -K dst IP_YOUR_AGENT_USES_FOR_CLOUD_CONNECTION`

Testing properties parser for datatypes which EMQX doesn't actually use can be done using:
`mosquitto_pub -t "test" -m "umbakarna" --property PUBLISH user-property "daewjaedon" "maybe"` while running mqtt_websockets test application. This way all of the parser options can be verified even though cloud actually doesn't use them.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Comment by thiagoftsm** [comment]
Hello @underhood ,

It is not necessary to bring nothing for this PR, but when you have a time, please, take a look at:

```sh
mqtt_websockets/src/mqtt_wss_client.c: In function 'http_proxy_connect':
mqtt_websockets/src/mqtt_wss_client.c:587:5: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
  587 |     write(client->sockfd, r_buf_ptr, strlen(r_buf_ptr));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mqtt_websockets/src/mqtt_wss_client.c:620:9: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
  620 |         write(client->sockfd, r_buf_ptr, strlen(r_buf_ptr));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mqtt_websockets/src/mqtt_wss_client.c:623:5: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
  623 |     write(client->sockfd, HTTP_ENDLINE, strlen(HTTP_ENDLINE));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mqtt_websockets/src/mqtt_wss_client.c: In function 'util_clear_pipe':
mqtt_websockets/src/mqtt_wss_client.c:1031:11: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' [-Wunused-result]
 1031 |     (void)read(fd, throwaway, THROWAWAY_BUF_SIZE);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mqtt_websockets/src/mqtt_wss_client.c: In function 'mqtt_wss_wakeup':
mqtt_websockets/src/mqtt_wss_client.c:1024:5: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
 1024 |     write(client->write_notif_pipe[PIPE_WRITE_END], " ", 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Best regards!

**Comment by underhood** [comment]
@thiagoftsm creating PR to fix this warnings

**Comment by thiagoftsm** [comment]
Yesterday I ran this PR during few hours and I did not observe errors, today I am going deeper with tests, so I will copy part of them here to avoid a giant text during the  last step (approve).

The `ss` is given expected results:

```sh
root@hades:~# netstat -antp | grep netdata
tcp        0      0 127.0.0.1:8125          0.0.0.0:*               LISTEN      12118/netdata       
tcp        0      0 127.0.0.1:8125          0.0.0.0:*               LISTEN      12118/netdata       
tcp        0      0 0.0.0.0:19999           0.0.0.0:*               LISTEN      12118/netdata       
tcp        0      0 192.168.0.36:60596      44.207.131.212:443      ESTABLISHED 12118/netdata       
tcp6       0      0 :::19999                :::*                    LISTEN      12118/netdata       
root@hades:~# ss -K dst 44.207.131.212:443
Netid                   State                   Recv-Q                   Send-Q                                     Local Address:Port                                        Peer Address:Port                   Process                   
tcp                     ESTAB                   0                        0                                           192.168.0.36:60596                                     44.207.131.212:https                            
```

**Comment by thiagoftsm** [comment]
Hello @underhood,

when I tried to compile our tests, I am observing these errors:

```sh
bash-5.2# mkdir testing
bash-5.2# cd testing/
bash-5.2# cmake ..
-- The C compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/thiago/Netdata/tests_netdata/mqtt_websockets/MQTT-C/testing
bash-5.2# make all
[ 11%] Building C object CMakeFiles/mqttc.dir/src/mqtt_pal.c.o
[ 22%] Building C object CMakeFiles/mqttc.dir/src/mqtt.c.o
/home/thiago/Netdata/tests_netdata/mqtt_websockets/MQTT-C/src/mqtt.c: In function ‘mqtt_mq_realloc’:
/home/thiago/Netdata/tests_netdata/mqtt_websockets/MQTT-C/src/mqtt.c:1660:5: error: unknown type name ‘uintptr_t’
 1660 |     uintptr_t curr_relative = mq->curr - (uint8_t *)mq->mem_start;
```

Except for this issue to run `mosquitto_pub`, all other tests passed with success.

**Comment by underhood** [comment]
@thiagoftsm we will be removing that code soon https://github.com/underhood/mqtt_websockets/pull/15 the old MQTT-C implementation is not possible to use in netdata anymore and we use only our new custom written MQTT 5 impl. now.

That is why that code will be removed.

**Comment by underhood** [comment]
@thiagoftsm after your review I decide to merge the changes on submodule https://github.com/underhood/mqtt_websockets/pull/14

last commit therefore just change pointer from that PR to master of submodule without changing any code

**Review by thiagoftsm — APPROVED** [review]
After @underhood explanation I retest PR and I got expected results, LGTM!

**Comment by underhood** [comment]
@MrZammler @stelfrag did you have time to take a look / test?

**Comment by MrZammler** [comment]
Hi @underhood testing it, will let you know.

**Review by MrZammler — APPROVED** [review]
Haven't noticed any issues.

**Pull request merged**
